security-analysis/security-analysis-api/src/main/java/com/powsybl/security/SecurityAnalysisTool.java

```diff
@@ -155,23 +158,46 @@ protected TableFormatterConfig createTableFormatterConfig() {
     }
 
+    private static void copyBytesToPath(byte[] bytes, Path dest) {
+        try (ByteArrayInputStream bis = new ByteArrayInputStream(bytes);
+             OutputStream fos = Files.newOutputStream(dest)) {
+            IOUtils.copy(bis, fos);
+        } catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
+    }
+
     private static SecurityAnalysisResult runSecurityAnalysis(CommandLine line, ToolRunningContext context, ContingenciesProvider contingenciesProvider, SecurityAnalysisParameters parameters, SecurityAnalysis securityAnalysis, String currentState) {
         SecurityAnalysisResult result;
         if (!line.hasOption(OUTPUT_LOG_OPTION)) {
             result = securityAnalysis.run(currentState, parameters, contingenciesProvider).join();
         } else {
-            SecurityAnalysisResultWithLog resultWithLog = securityAnalysis.runWithLog(currentState, parameters, contingenciesProvider).join();
-            result = resultWithLog.getResult();
-            // copy log bytes to file
-            resultWithLog.getLogBytes().ifPresent(logBytes -> {
-                Path outlogDest = context.getFileSystem().getPath(line.getOptionValue(OUTPUT_LOG_OPTION));
-                try (ByteArrayInputStream bis = new ByteArrayInputStream(logBytes);
-                     OutputStream fos = Files.newOutputStream(outlogDest)) {
-                    IOUtils.copy(bis, fos);
-                } catch (IOException e) {
-                    throw new UncheckedIOException(e);
+            try {
+                SecurityAnalysisResultWithLog resultWithLog = securityAnalysis.runWithLog(currentState, parameters, contingenciesProvider).join();
+                result = resultWithLog.getResult();
+                // copy log bytes to file
+                resultWithLog.getLogBytes().ifPresent(logBytes -> {
+                    Path outlogDest = context.getFileSystem().getPath(line.getOptionValue(OUTPUT_LOG_OPTION));
+                    copyBytesToPath(logBytes, outlogDest);
+                });
+            } catch (CompletionException e) {
+                if (e.getCause() instanceof ComputationException) {
+                    ComputationException computationException = (ComputationException) e.getCause();
+                    Map<String, String> merged = new HashMap<>(computationException.getOutLogs());
+                    merged.putAll(computationException.getErrLogs());
+                    Map<String, byte[]> bytesByName =
+                            merged.entrySet().stream()
+                                    .collect(Collectors.toMap(
+                                            Map.Entry::getKey,
+                                        entry -> entry.getValue().getBytes()
+                                    ));
+                    bytesByName.putAll(computationException.getZipBytes());
+                    byte[] bytes = ZipHelper.archiveBytesByNameToZipBytes(bytesByName);
```

Comment:
I think the exception could provide this method to create this zip containing all files, so that it can be used by other client codes.
Maybe this code could be shared with the code which creates the zip for the SecurityAnalysisResultWithLog.
(actually, we can wonder if the exception should not contain directly this zip instead of the list of all files, but it's ok for me to have more details :) )

Reply:
@yichen : could you just add this code to a method in computation exception ?
We will need to re-use it in client code which wants to collect all logs in one zip.

Reply:
As just discussed, this can be part of another PR (maybe with a zip builder instead of the static methods in zip helper).